source/server/lds_api.cc

```diff
@@ -28,8 +29,14 @@ LdsApiImpl::LdsApiImpl(const envoy::config::core::v3::ConfigSource& lds_config,
       listener_manager_(lm), scope_(scope.createScope("listener_manager.lds.")), cm_(cm),
       init_target_("LDS", [this]() { subscription_->start({}); }) {
   const auto resource_name = getResourceName();
-  subscription_ = cm.subscriptionFactory().subscriptionFromConfigSource(
-      lds_config, Grpc::Common::typeUrl(resource_name), *scope_, *this, resource_decoder_);
+  if (lds_resources_locator.ByteSize() == 0) {
```

Comment:
This seems like an odd/fragile check. Should this be passe by const pointer or optional reference?